holoviews/plotting/renderer.py

```diff
@@ -147,6 +147,8 @@ def get_plot(self_or_cls, obj):
         # Initialize DynamicMaps with first data item
         dmaps = obj.traverse(lambda x: x, specs=[DynamicMap])
         for dmap in dmaps:
+            if dmap.sampled:
+                continue
```

Comment:
Maybe add a comment to explain it is initialized in the plotting code itself (because that is where you know the sample keys).